packages/mobx/__tests__/v5/base/typescript-tests.ts

```diff
@@ -2055,6 +2055,23 @@ test("type inference of the action callback", () => {
     }
 })
 
+test("TS - type inference of reaction opts.equals", () => {
+    const data = observable({ a: 23 })
+    mobx.reaction(
+        () => data.a,
+        a => {},
+        {
+            equals: (a, b) => {
+                assert<IsExact<typeof a, string>>(false)
+                assert<IsExact<typeof b, string>>(false)
```

Comment:
If you don't add these two lines, it seems like TS would just converge the anys into `number` and make the test pass even without the changes below.